packages/table/src/table.tsx

```diff
@@ -952,116 +953,104 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
-                case RegionCardinality.FULL_TABLE:
-                    style.left = "-1px";
-                    style.borderLeft = "none";
-                    style.bottom = "-1px";
-                    style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
-                    return style;
-                case RegionCardinality.FULL_COLUMNS:
-                    style.bottom = "-1px";
-                    style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
-                    return style;
-
-                default:
-                    return { display: "none" };
-            }
-        };
-        return this.maybeRenderRegions(styler);
+    private styleColumnHeaderRegion = (region: IRegion): React.CSSProperties => {
+        const { grid } = this;
+        const { viewportRect } = this.state;
+        if (viewportRect == null) {
+            return {};
+        }
+        const cardinality = Regions.getRegionCardinality(region);
+        const style = grid.getRegionStyle(region);
+
+        switch (cardinality) {
+            case RegionCardinality.FULL_TABLE:
+                style.left = "-1px";
+                style.borderLeft = "none";
+                style.bottom = "-1px";
+                style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
+                return style;
+            case RegionCardinality.FULL_COLUMNS:
+                style.bottom = "-1px";
+                style.transform = `translate3d(${-viewportRect.left}px, 0, 0)`;
+                return style;
+
+            default:
+                return { display: "none" };
+        }
     }
 
-    private maybeRenderRowHeaderRegions() {
-        const styler = (region: IRegion): React.CSSProperties => {
-            const { grid } = this;
-            const { viewportRect } = this.state;
-            if (viewportRect == null) {
-                return {};
-            }
-            const cardinality = Regions.getRegionCardinality(region);
-            const style = grid.getRegionStyle(region);
-            switch (cardinality) {
-                case RegionCardinality.FULL_TABLE:
-                    style.top = "-1px";
-                    style.borderTop = "none";
-                    style.right = "-1px";
-                    style.transform = `translate3d(0, ${-viewportRect.top}px, 0)`;
-                    return style;
-                case RegionCardinality.FULL_ROWS:
-                    style.right = "-1px";
-                    style.transform = `translate3d(0, ${-viewportRect.top}px, 0)`;
-                    return style;
-
-                default:
-                    return { display: "none" };
-            }
-        };
-        return this.maybeRenderRegions(styler);
+    private styleRowHeaderRegion = (region: IRegion): React.CSSProperties => {
+        const { grid } = this;
+        const { viewportRect } = this.state;
+        if (viewportRect == null) {
+            return {};
+        }
```

Comment:
this `viewportRect` check seems to happen in every one of these functions. maybe it can be moved up a level to the function that calls these `style*Region()` functions?